libs/api/domains/license-service/src/lib/providers/tokenService.provider.ts

```diff
@@ -0,0 +1,12 @@
+import { ConfigType } from '@island.is/nest/config'
+import { FactoryProvider } from '@nestjs/common'
+import { LicenseServiceConfig } from '../licenseService.config'
+import { TOKEN_SERVICE_PROVIDER } from '../licenseService.constants'
+import { TokenService } from '../services/token.service'
+
+export const TokenServiceProvider: FactoryProvider = {
+  provide: TOKEN_SERVICE_PROVIDER,
+  useFactory: (licenseServiceConfig: ConfigType<typeof LicenseServiceConfig>) =>
+    new TokenService(licenseServiceConfig.barcodeSecretKey),
+  inject: [LicenseServiceConfig.KEY],
+}
```

Comment:
~I'm conflicted on having simple provider like this in its own file rather than colocated with the service or just reading straight from the config in the TokenService constructor. It feels like an extra level of indirection which arguably makes it more difficult to understand the lib.

Reply:
See
https://github.com/island-is/island.is/pull/13984#discussion_r1514695565